File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        let result = [];
        if(nums.length === 0) {
            return result
        }
        for(let i = 0; i<nums.length; i++) {
            if(i === 0) {
                result.push(nums.slice(1, nums.length))
            } else if(i === nums.length - 1) {
                result.push(nums.slice(0, nums.length-1))
            } else {
                const arr1 = nums.slice(0, i)
                const arr2 = nums.slice(i+1, nums.length)
                result.push([...arr1, ...arr2])
            }
        }

        return result.map(n => n.reduce((t, c) => t*c, 1))
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] result = new int[nums.length * 2];
        for(int i = 0; i < nums.length * 2; i++) {
            if(i < nums.length) {
                result[i] = nums[i];
            } else {
                result[i] = nums[i - nums.length];
            }
        }

        return result;
    }
}